plugins/blockvolume/blockprovider/gluster-loopback/glusterloopback.go

```diff
@@ -0,0 +1,274 @@
+package glusterloopback
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/plugins/blockvolume/blockprovider"
+	"github.com/gluster/glusterd2/plugins/blockvolume/utils"
+
+	log "github.com/sirupsen/logrus"
+	config "github.com/spf13/viper"
+)
+
+const providerName = "gluster-loopback"
+
+func init() {
+	log.WithField("name", providerName).Infof("Registering block provider")
+	blockprovider.RegisterBlockProvider(providerName, newGlusterBlock)
+}
+
+// GlusterBlock implements block Provider interface. It represents a gluster-block
+type GlusterBlock struct {
+	mounts map[string]string
+}
+
+func newGlusterBlock() (blockprovider.Provider, error) {
+	gb := &GlusterBlock{}
+
+	gb.mounts = make(map[string]string)
+
+	return gb, nil
+}
+
+func mountHost(g *GlusterBlock, hostVolume string) (string, error) {
+	hostDir := g.mounts[hostVolume]
+	if hostDir == "" {
+		//Mount the host volume if not already mounted
+		hostDir := config.GetString("rundir") + "/" + hostVolume
+		err := os.MkdirAll(hostDir, os.ModeDir|os.ModePerm)
+		if err != nil {
+			return "", err
+		}
+
+		err = volume.MountVolume(hostVolume, hostDir)
+		if err != nil {
+			return "", err
+		}
+		g.mounts[hostVolume] = hostDir
+	}
+	return hostDir, nil
+}
+
+// CreateBlockVolume will create a gluster block volume with given name and size having `hostVolume` as hosting volume
+func (g *GlusterBlock) CreateBlockVolume(name string, size uint64, hosts []string, hostVolume string, options ...blockprovider.BlockVolOption) (blockprovider.BlockVolume, error) {
+	var (
+		blockVolOpts = &blockprovider.BlockVolumeOptions{}
+		clusterLocks = transaction.Locks{}
+	)
+
+	blockVolOpts.ApplyOpts(options...)
+
+	if err := clusterLocks.Lock(hostVolume); err != nil {
+		log.WithError(err).Error("error in acquiring cluster lock")
+		return nil, err
+	}
+	defer clusterLocks.UnLock(context.Background())
+
+	volInfo, err := volume.GetVolume(hostVolume)
```

Comment:
https://github.com/gluster/glusterd2/pull/1357#discussion_r244920282 fixing this helps to reduce duplicate code here.